deep_agent/aegra/mcp.py

```
@@ -683,9 +690,10 @@ async def get_mcp_tools(
             logger.warning("MCP tools deferred — no auth token at startup")
         return []
 
-    _cached_tools = tools
-    _cached_tools_ts = time.time()
+    _cached_tools[cache_key] = tools
+    _cached_tools_ts[cache_key] = time.time()
     logger.warning(
-        f"Loaded {len(tools)} MCP tool(s): {', '.join(seen)} (cached for {_MCP_TOOL_CACHE_TTL:.0f}s)"
+        f"Loaded {len(tools)} MCP tool(s): {', '.join(seen)} "
+        f"(cached for {_MCP_TOOL_CACHE_TTL:.0f}s, user={cache_key or 'anonymous'})"
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)